airflow/utils/decorators.py

```diff
@@ -21,13 +21,16 @@
 import os
 from copy import copy
 from functools import wraps
+from typing import Callable, TypeVar, cast
 
 from airflow.exceptions import AirflowException
 
 signature = inspect.signature
 
+T = TypeVar('T', bound=Callable)  # pylint: disable=invalid-name
```

Comment:
This is the most interesting part.

Reply:
Not sure I understand it. I wondering if removing the `@apply_defulats` will solve the same issue? I think now we have `BaseOperatorMeta` we can consider removing this decorator.

Reply:
I will solve it because mypy is not smart enough to look into the metaclass, but we will still have remaining problems to solve. I consider it a temporary solution until we cannot use the metaclass .